src/strategies/java.ts

```diff
@@ -0,0 +1,280 @@
+        breaking: false,
+        scope: null,
+        notes: [],
+        files: [],
+        references: [],
+        sha: 'fake',
+      });
+    }
+    return commits;
+  }
+
+  private static isSnapshot(version?: Version): boolean {
+    return !!version?.preRelease && version.preRelease.indexOf('SNAPSHOT') >= 0;
+  }
+
+  protected async needsSnapshot(
+    commits: Commit[],
+    latestRelease?: Release
+  ): Promise<boolean> {
+    const version = latestRelease?.tag?.version;
+    if (!version) {
+      // Don't bump snapshots for first release ever
+      return false;
+    }
+
+    // Found snapshot as a release, this is unexpected, but use it
+    if (Java.isSnapshot(version)) {
+      return false;
+    }
+
+    // Search commits for snapshot bump
+    const snapshotCommits = commits
+      .map(commit =>
+        PullRequestTitle.parse(
+          commit.pullRequest?.title || commit.message,
+          this.pullRequestTitlePattern
+        )
+      )
+      .map(pullRequest => pullRequest?.getVersion())
+      .filter(Java.isSnapshot);
+    return snapshotCommits.length === 0;
+  }
+
+  isValidRelease(version: Version): boolean {
+    return !version.preRelease || version.preRelease.indexOf('SNAPSHOT') < 0;
+  }
+
+  protected async buildUpdates(
+    options: JavaBuildUpdatesOption
+  ): Promise<Update[]> {
+    const updates: Update[] = [];
+
+    const pomFilesSearch = this.github.findFilesByFilenameAndRef(
+      'pom.xml',
+      this.targetBranch,
+      this.path
+    );
+    const buildFilesSearch = this.github.findFilesByFilenameAndRef(
+      'build.gradle',
+      this.targetBranch,
+      this.path
+    );
+    const dependenciesSearch = this.github.findFilesByFilenameAndRef(
+      'dependencies.properties',
+      this.targetBranch,
+      this.path
+    );
+
+    const allFiles = await Promise.all([
+      pomFilesSearch,
+      buildFilesSearch,
+      dependenciesSearch,
+    ]).then(result => result.flat());
+    allFiles.forEach(path => {
+      this.buildFileUpdates(updates, path, options);
+    });
+
+    if (!options.isSnapshot) {
+      this.extraFiles.forEach(extraFile => {
+        if (typeof extraFile === 'object') {
+          return;
+        }
+        this.buildFileUpdates(updates, extraFile, options, true);
+      });
+
+      updates.push({
+        path: this.addPath(this.changelogPath),
+        createIfMissing: true,
+        updater: new Changelog({
+          version: options.newVersion,
+          changelogEntry: options.changelogEntry,
+        }),
+      });
+    }
+
+    return updates;
+  }
+
+  protected buildFileUpdates(
+    updates: Update[],
+    path: string,
+    options: JavaBuildUpdatesOption,
+    extraFile?: boolean
+  ) {
+    if (!extraFile) {
+      updates.push({
+        path: this.addPath(path),
+        createIfMissing: false,
+        updater: new Generic({
+          version: options.newVersion,
+          versionsMap: options.versionsMap,
+        }),
+      });
+    }
```

Comment:
Sorry, I'm not following why we're skipping the file if it's an `extraFile`

Reply:
Resolved by refactoring.